changes to Thread controller methods and routes

bHarlum opened the pull request:
Completed all controller methods,
Removed make and edit route as they are not needed for a backend api.

bHarlum committed:
thread routes/controller

 - DRYING: Removed edit routes and controller function as they return the same data as the show route and function. 
- NEW: Updated functions to be 'async'.
- NEW: Added logic for the index and show functions.

bHarlum committed:
Threads controller: update function

NEW: Added logic to update function.

bHarlum committed:
thread route/controller

Removed make route and function

bHarlum committed:
Thread controller

NEW: Added logic to create method.

bHarlum committed:
thread Controller: create, update

Added comments to create and update functions.

bHarlum committed:
Thread controller

NEW: Added logic to destroy function.
NEW: Added a new function for generating error response.

bHarlum committed:
thread controller

FIX: Using create function off of the model instead of insertOne as insertOne is only on collections.

bHarlum committed:
Merge branch 'dev' into thread-routes

ClementineOldfield reviewed (changes requested):
Change thread_controller method

bHarlum committed:
fixed logic error in threads controller

changed if statement within update function and added else to fix a logic error.
